omnibus/files/private-chef-cookbooks/private-chef/libraries/preflight_bootstrap_validator.rb

```diff
@@ -75,10 +75,11 @@ def run!
   # backend components, allow data bootstrapping to be bypassed when
   # no chef-server-running.json is present but a secrets file is present.
   def bypass_bootstrap?
-    first_run? && secrets_exists?
+    first_run? && secrets_exists? && PrivateChef['postgresql']['external']
```

Comment:
I definitely think that's a much better way to fix this, but I had to add this check otherwise copying the secrets file between nodes on existing HA nodes would fail.